File: examples/nodejs/streaming.js
/**
 * 蓝鹰AI网关 - Node.js 流式输出示例
 * BlueEagle AI Gateway - Node.js Streaming Example
 *
 * 官方网站: https://ahg.codes
 */

import OpenAI from 'openai';

const client = new OpenAI({
  apiKey: 'YOUR_API_KEY',
  baseURL: 'https://ahg.codes/v1',
});

async function streamChat(model = 'gpt-4o', prompt = '用中文写一首关于蓝鹰的诗') {
  try {
    console.log(`模型: ${model}`);
    console.log(`问题: ${prompt}`);
    console.log('-'.repeat(50));
    process.stdout.write('回复: ');

    const stream = await client.chat.completions.create({
      model,
      messages: [
        { role: 'system', content: '你是一个有创意的AI助手。' },
        { role: 'user', content: prompt },
      ],
      stream: true,
      temperature: 0.8,
      max_tokens: 2048,
    });

    let fullResponse = '';

    for await (const chunk of stream) {
      const content = chunk.choices[0]?.delta?.content;
      if (content) {
        process.stdout.write(content);
        fullResponse += content;
      }
    }

    console.log('\n' + '-'.repeat(50));
    console.log(`完整回复长度: ${fullResponse.length} 字符`);
  } catch (error) {
    console.error('请求失败:', error.message);
  }
}

// 执行示例
(async () => {
  await streamChat('gpt-4o');
  console.log('\n');
  await streamChat('claude-4-sonnet', 'Explain quantum computing in simple terms');
})();

File: examples/python/streaming.py
"""
蓝鹰AI网关 - Python 流式输出示例
BlueEagle AI Gateway - Python Streaming Example

官方网站: https://ahg.codes
"""

from openai import OpenAI

client = OpenAI(
    api_key="YOUR_API_KEY",
    base_url="https://ahg.codes/v1"
)

def stream_chat(model: str = "gpt-4o", prompt: str = "用中文写一首关于蓝鹰的诗"):
    """流式输出对话"""

    print(f"模型: {model}")
    print(f"问题: {prompt}")
    print("-" * 50)
    print("回复: ", end="", flush=True)

    stream = client.chat.completions.create(
        model=model,
        messages=[
            {"role": "system", "content": "你是一个有创意的AI助手。"},
            {"role": "user", "content": prompt}
        ],
        stream=True,
        temperature=0.8,
        max_tokens=2048
    )

    full_response = ""
    for chunk in stream:
        if chunk.choices and chunk.choices[0].delta.content:
            content = chunk.choices[0].delta.content
            print(content, end="", flush=True)
            full_response += content

    print("\n" + "-" * 50)
    print(f"完整回复长度: {len(full_response)} 字符")


if __name__ == "__main__":
    # GPT-4o 流式输出
    stream_chat(model="gpt-4o")

    print("\n")

    # Claude 流式输出
    stream_chat(model="claude-4-sonnet", prompt="Explain quantum computing in simple terms")
